layers/meta-balena-jetson/recipes-bsp/tegra-binaries/tegra186-flash-dry_32.1.0.bb

```diff
@@ -0,0 +1,191 @@
+SUMMARY = "Create flash artifacts without flashing"
+
+LICENSE = "Apache-2.0"
+LIC_FILES_CHKSUM = "file://${RESIN_COREBASE}/COPYING.Apache-2.0;md5=89aea4e17d99a7cacdbeed46a0096b10"
+
+DEPENDS = " \
+    coreutils-native \
+    virtual/bootloader \
+    virtual/kernel \
+    tegra-binaries \
+    tegra-bootfiles \
+    tegra186-flashtools-native \
+    dtc-native \
+ "
+
+inherit deploy pythonnative perlnative
+
+SRC_URI = " \
+    file://resinOS-flash186.xml \
+    file://partition_specification186.txt \
+"
+
+IMAGE_UBOOT ??= "u-boot-dtb"
+
+LNXFILE="u-boot-dtb.bin"
+IMAGE_TEGRAFLASH_KERNEL="${DEPLOY_DIR_IMAGE}/${LNXFILE}"
+DTBFILE ?= "${@os.path.basename(d.getVar('KERNEL_DEVICETREE', True).split()[0])}"
+
+LDK_DIR = "${TMPDIR}/work-shared/L4T-${SOC_FAMILY}-${PV}-${PR}/Linux_for_Tegra"
+B = "${WORKDIR}/build"
+S = "${WORKDIR}"
+
+BINARY_INSTALL_PATH = "/opt/tegra-binaries"
+
+BOOTFILES=" \
+    adsp-fw.bin \
+    bmp.blob \
+    bpmp.bin \
+    camera-rtcpu-sce.img \
+    dram-ecc.bin \
+    eks.img \
+    mb1_prod.bin \
+    mb1_recovery_prod.bin \
+    mce_mts_d15_prod_cr.bin \
+    nvtboot_cpu.bin \
+    nvtboot_recovery.bin \
+    nvtboot_recovery_cpu.bin \
+    preboot_d15_prod_cr.bin \
+    slot_metadata.bin \
+    spe.bin \
+    tos-mon-only.img \
+    nvtboot.bin \
+    warmboot.bin \
+    minimal_scr.cfg \
+    mobile_scr.cfg \
+    emmc.cfg \
+"
+
+BINS="mb2_bootloader nvtboot_recovery.bin; \
+mts_preboot preboot_d15_prod_cr.bin; \
+mts_bootpack mce_mts_d15_prod_cr.bin; \
+bpmp_fw bpmp.bin; \
+bpmp_fw_dtb tegra186-a02-bpmp-quill-p3310-1000-c04-00-te770d-ucm2.dtb; \
+tlk tos-mon-only.img; \
+eks eks.img; \
+bootloader_dtb $DTBFILE;"
+
+do_configure() {
+    local f
+
+    sed -i -e "s/\[DTB_NAME\]/$(echo ${DTBFILE} | cut -d '.' -f 1)/g" ${WORKDIR}/partition_specification186.txt
+    if [ -d ${DEPLOY_DIR_IMAGE}/bootfiles ]; then
+         rm -rf ${DEPLOY_DIR_IMAGE}/bootfiles
+    fi;
+
+    if [ -d ${DEPLOY_DIR_IMAGE}/tegra-binaries ]; then
+         rm -rf ${DEPLOY_DIR_IMAGE}/tegra-binaries
+    fi;
+
+    mkdir -p ${DEPLOY_DIR_IMAGE}/bootfiles
+    local f
+    PATH="${STAGING_BINDIR_NATIVE}/tegra186-flash:${PATH}"
+    rm -rf "${WORKDIR}/tegraflash"
+    mkdir -p "${WORKDIR}/tegraflash"
+    oldwd=`pwd`
+    cd "${WORKDIR}/tegraflash"
+    ln -s "${STAGING_DATADIR}/tegraflash/${MACHINE}.cfg" .
+    ln -s "${IMAGE_TEGRAFLASH_KERNEL}" ./${LNXFILE}
```

Comment:
final u-boot image written to kernel & kernel_b is generated by mkbootimg. we need a dependency for do_compile() for that task too.